src/networking.c

```diff
@@ -209,6 +210,91 @@ cc_oci_tap_create(const gchar *const tap) {
 }
 
 /*!
+ * Helper function for setting/getting MTU for network interface.
+ *
+ * \param ifname Interface name.
+ * \param mtu MTU to be set/retrieved for the interface.
+ * \param set Flag indicating if the mtu needs to be set/retrieved.
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+cc_oci_handle_interface_mtu(char *ifname, unsigned int *mtu, gboolean set)
+{
+	int socket_fd = -1;
+	gboolean ret = false;
+	struct ifreq ifr;
+
+	if (! ifname) {
+		return false;
+	}
+
+	socket_fd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (socket_fd < 0) {
+		g_critical("Error opening socket: %s", strerror(errno));
+		goto out;
+	}
+
+	ifr.ifr_addr.sa_family = AF_INET;
+	g_strlcpy(ifr.ifr_name, ifname, IFNAMSIZ);
+
+	if (set) {
+		ifr.ifr_mtu = (int)(*mtu);
+
+		if (ioctl(socket_fd, SIOCSIFMTU, (caddr_t)&ifr) < 0) {
+			g_critical("ioctl SIOCSIFMTU (set mtu) error for interface %s" \
+					"with mtu %d :%s", 
+					ifname, *mtu, strerror(errno));
+			goto out;
+		}
+		g_debug("MTU set for interface %s as %d\n", ifname, ifr.ifr_mtu);
+	} else {
+		if (ioctl(socket_fd, SIOCGIFMTU, (caddr_t)&ifr) < 0) {
+			g_critical("ioctl get mtu error for interface %s :%s",
+				ifname, strerror(errno));
+			goto out;
+		}
+		*mtu = (unsigned int)ifr.ifr_mtu;
+		g_debug("MTU for interface %s: %d\n", ifname, *mtu);
+	}
+
+	ret = true;
+out:
+	if (socket_fd > 0) {
```

Comment:
technically, this check should probably `>=0`, seeing as 0 is a valid fd (although the odds of you being handed it are somewhat minimal). Given you set the default value to -1 at the top, probably change this to a `<0` ?